File: backend/api/connections.py
from fastapi import APIRouter, HTTPException
from models import ConnectionRequest, ConnectionResponse
from services import connection_service

router = APIRouter()

@router.post("/find", response_model=ConnectionResponse)
async def find_connections(request: ConnectionRequest):
    """Find connections for selected text across all documents"""
    try:
        response = connection_service.find_connections(
            selected_text=request.selected_text,
            current_doc_id=request.current_document_id,
            context_before=request.context_before,
            context_after=request.context_after
        )
        return response
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: frontend/src/services/api/endpoints/connections.js
/**
 * Cross-document connections API
 * Finds relationships and connections between different documents
 */
import { api } from '../config';
import { handleApiError } from '../utils/errorHandler';
import { toStringSafe, toNumberSafe } from '../utils/validators';
import { transformConnection } from '../utils/transformers';

export const findConnections = async ({ 
  selected_text, 
  current_document_id, 
  current_page, 
  context_before = '', 
  context_after = '' 
}) => {
  try {
    const payload = {
      selected_text: toStringSafe(selected_text),
      current_document_id: toStringSafe(current_document_id),
      current_page: toNumberSafe(current_page, 1),
      context_before: toStringSafe(context_before),
      context_after: toStringSafe(context_after),
    };

    if (!payload.selected_text?.trim()) {
      throw new Error('No selected text provided');
    }
    if (!payload.current_document_id?.trim()) {
      throw new Error('Missing document identifier');
    }

    const response = await api.post('/api/connections/find', payload);
    const data = response?.data || {};

    const normalized = {
      connections: Array.isArray(data.connections) 
        ? data.connections.map(transformConnection) 
        : [],
      summary: toStringSafe(data.summary),
      processing_time: Number(data.processing_time) || 0,
    };

    console.log('Connections response:', normalized);
    return normalized;
  } catch (error) {
    throw new Error(handleApiError(error, 'Failed to fetch connections'));
  }
};
